src/main/java/org/opentripplanner/routing/impl/OrcaFareServiceImpl.java

```diff
@@ -230,27 +237,32 @@ public boolean populateFare(Fare fare,
                                    List<Ride> rides,
                                    Collection<FareRuleSet> fareRules
     ) {
-        long freeTransferTripTime = 0;
+        Long freeTransferStartTime = null;
         float cost = 0;
         float orcaFareDiscount = 0;
         for (Ride ride : rides) {
             RideType rideType = classify(ride.routeData);
-            freeTransferTripTime += ride.endTime - ride.startTime;
+            if (freeTransferStartTime == null && permitsFreeTransfers(rideType)) {
+                // The start of a free transfer must be with a transit agency that permits it!
+                freeTransferStartTime = ride.startTime;
+            }
             float singleLegPrice = getRidePrice(ride, fareType, fareRules);
             float discountedFare = getDiscountedFare(fareType, rideType, singleLegPrice);
-            if (hasFreeTransfers(fareType, rideType, freeTransferTripTime)) {
+            if (hasFreeTransfers(fareType, rideType) && inFreeTransferWindow(freeTransferStartTime, ride.startTime)) {
                 // If using Orca (free transfers), the total fare should be equivalent to the
                 // most expensive leg of the journey.
                 orcaFareDiscount = Float.max(orcaFareDiscount, discountedFare);
             } else {
                 // If free transfers are not permitted, add the cash price of this leg to the total cost.
-                // This case is for Washington State Ferries, which do not offer any discounts.
                 cost += singleLegPrice;
             }
-            if (freeTransferTripTime > FREE_TRANSFER_TIME_LIMIT) {
+            if (!inFreeTransferWindow(freeTransferStartTime, ride.startTime)) {
```

Comment:
Should this if block be at the beginning of the for loop block, and contain the free transfer window init?

Reply:
Hmm I'm not sure it should. I need to keep the init and the reset (`freeTransferStartTime = null`) separate so that the cash fare after the free transfer window is applied. Moving it to the begining does the reset of the free transfer window, but does not apply the cash fare. Perhaps my understanding is not correct?

Reply:
I think so yes.